pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -377,16 +377,14 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 			}
 		}
 
-		if len(in.AppendHeaders) > 0 {
-			action.RequestHeadersToAdd = make([]*core.HeaderValueOption, 0)
-			for key, value := range in.AppendHeaders {
-				action.RequestHeadersToAdd = append(action.RequestHeadersToAdd, &core.HeaderValueOption{
-					Header: &core.HeaderValue{
-						Key:   key,
-						Value: value,
-					},
-				})
-			}
+		// TODO: deprecate this
```

Comment:
why>

Reply:
The question is do we need these functions in both HTTPRoute (applies to all destination) and DestinationWeight (applies to only that destination)?

If we don't deprecate this one (and also removeResponseHeaders), we should probably add the other 2 (appendResponse and removeRequest), so that we have all 4 append/remove fields at this level too.

Reply:
Can we make it exclusive ? Either top level or internal ?

Additionally, the Envoy semantics is that top level ones are added first, then the ones in per destination are added on top of it. So theoretically I could do 
 Add abc for all
 Add xyz for v1 (in addition to abc).